### ISSUE TITLE
Rule: no-return-await with try...catch

### ISSUE THREAD

**Issue by travisperson** [eslint/eslint.github.io#370]
I don't know if this rule needs to be updated, or if just further clarification in the documentation is required.

Rule: http://eslint.org/docs/rules/no-return-await

When using a `try...catch` in a `async` function, to unwrap the promise and catch an error you must await the promise.

```javascript
async function foo() {
  try {
    return await bar();
  } catch (err) {
    // handle err
  }
}
```

Try: https://repl.it/Jslh
```javascript
function rejector () {
  return new Promise((resolve, reject) => {
    setTimeout(() => reject(new Error('Failure')), 1000);
  });
}


async function tryCatcher (promise, awaitPromise = false) {
  try {
    return promise;
  } catch (err) {
    console.error('Catched in tryCatcher', err)
  }
}

async function test () {
  const promise = rejector();

  try {
    await tryCatcher(promise);
  } catch (err) {
    console.error('Catched in main', err)
  }

}

test();

```

**Comment by not-an-aardvark** [comment]
The rule already handles this case correctly (it does not report an error if the `await` is in a `try` statement). I agree that it might be good to add an example of this to the documentation -- would you like to make a PR?

Note that the file should be modified in `eslint/eslint` [here](https://github.com/eslint/eslint/blob/78a85e0a255ec93ecf71c20afa53facb992f8202/docs/rules/no-return-await.md). The file in this repository is automatically built from the file in the `eslint/eslint` repository.

**Comment by Rekhyt** [comment]
I would like to stress that the **try-catch block** will become **useless** when omitting the `await`. I'd expect to be warned about this by ESLint.

This might be a new rule. Should I open a new issue for that?

**Comment by not-an-aardvark** [comment]
@Rekhyt Yes, please open a new issue. My initial impression is that the try-catch isn't useless there because it still catches synchronous errors (and ESLint can't detect what has the potential to throw a synchronous error), but I'm happy to talk about it more on a new issue.

**Comment by kaicataldo** [comment]
I'm going to go ahead and close this issue since the original conversation has stalled.

@Rekhyt Please feel free to make a new issue in the [eslint/eslint repository](https://github.com/eslint/eslint/issues).
